scipy/integrate/_quadrature.py

```diff
@@ -822,6 +825,271 @@ def simpson(y, *, x=None, dx=1.0, axis=-1, even=_NoValue):
+        `y`.
+    dx : scalar or array_like, optional
+        Spacing between elements of `y`. Only used if `x` is None. Can either 
+        be a float, or an array with the same shape as `y`, but length 1 along
+        `axis`.
+    axis : int, optional
+        Specifies the axis to integrate along. Default is -1 (last axis).
+    initial : scalar or array_like, optional
+        If given, insert this value at the beginning of the returned result,
+        and add it to the rest of the result. Default is None, which means no
+        value at ``x[0]`` is returned and `res` has one element less than `y`
+        along the axis of integration. Can either be a float, or an array with
+        the same shape as `y`, but length 1 along `axis`.
+
+    Returns
+    -------
+    res : ndarray
+        The result of cumulative integration of `y` along `axis`.
+        If `initial` is None, the shape is such that the axis of integration
+        has one less value than `y`. If `initial` is given, the shape is equal
+        to that of `y`.
+
+    See Also
+    --------
+    numpy.cumsum
+    cumulative_trapezoid : cumulative integration using the composite 
+        trapezoidal rule
+    simpson : integrator for sampled data using the Composite Simpson's Rule
+
+    Notes
+    -----
+
+    .. versionadded:: 1.12.0
+
+    The composite Simpson's 1/3 method can be used to approximate the definite 
+    integral of a sampled input function :math:`y(x)` [1]_. The method assumes 
+    a quadratic relationship over the interval containing any 3 consecutive 
+    sampled points. The integral is estimated separately for each interval
+    and then cumulatively summed to obtain the final result.
+
+    Consider three consecutive points: 
+    :math:`(x_{1}, y_{1}), (x_{2}, y_{2}), (x_{3}, y_{3})`.
+    where the widths of the 2 subintervals are:
+    :math:`h_{1} = x_{2} - x_{1}` and :math:`h_{2} = x_{3} - x_{2}`
+
+    Assuming a quadratic relationship over the 3 points, the following 
+    integrals for the subintervals are derived (see [2]_ for derivation 
+    method):
+
+    .. math::
+        \int_{x_{1}}^{x_{2}} y(x) dx\ = \int_{0}^{h_{1}} y(x) dx\
+            = \frac{h_{1}(2h_{1}+3h_{2})}{6(h_{1}+h_{2})} y_{1}
+            + \frac{h_{1}(h_{1}+3h_{2})}{6h_{2}} y_{2}
+            - \frac{h_{1}^{3}}{6h_{2}(h_{1}+h_{2})} y_{3}
+
+    .. math::
+        \int_{x_{2}}^{x_{3}} y(x) dx\ = \int_{h_{1}}^{h_{1}+h_{2}} y(x) dx\
+            = - \frac{h_{2}^{3}}{6h_{1}(h_{1}+h_{2})} y_{1}
+            + \frac{h_{2}(3h_{1}+h_{2})}{6h_{1}} y_{2}
+            + \frac{h_{2}(3h_{1}+2h_{2})}{6(h_{1}+h_{2})} y_{3}
+
+    .. math::
+        \int_{x_{1}}^{x_{3}} y(x) dx\ = \int_{0}^{h_{1}+h_{2}} y(x) dx\
+            = \frac{(h_{1}+h_{2})(2h_{1}-h_{2})}{6h_{1}} y_{1}
+            + \frac{(h_{1}+h_{2})^{3}}{6h_{1}h_{2}} y_{2}
+            - \frac{(h_{1}-2h_{2})(h_{1}+h_{2})}{6h_{2}} y_{3}
+
+    Note that the integral formulae in [1]_ and [2]_ use different variables.
```

Comment:
Let's replace all these with just Equation 8 from [2]. We can explain how the other integrals (from x2 to x3 and from x1 to x3) are calculated from that.

Reply:
Updated these equations to put them in the form of [2]. Just saw your comment. Since we are using the \int_{x_{2}}^{x_{3}} y(x) dx equation, I think it should be included.